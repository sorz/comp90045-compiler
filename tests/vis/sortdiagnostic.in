5 9 1 0 2 8 100 2
