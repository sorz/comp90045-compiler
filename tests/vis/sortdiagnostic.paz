{ this is the same as sort.paz but prints out what it is doing as it goes }
{ it may be useful if you are getting uninitialized value errors and so on }
program sortdiagnostic;

    var h, j, v: integer;
        x: array [0..8] of integer;

begin
    { we put a dummy sentinel value in the first element }
    { this is because short-circuit evaluation is not implemented }
    { thus when j reaches 0 it will read but ignore the x[0] entry }
    x[0] := 0;

    write('Provide 8 integers: ');
    read(x[1]);
    read(x[2]);
    read(x[3]);
    read(x[4]);
    read(x[5]);
    read(x[6]);
    read(x[7]);
    read(x[8]);

    for h := 2 to 8 do
    begin
        write('h=');
        write(h);
        write(' x[h]=');
        write(x[h]);
        writeln;
        v := x[h];
        j := h-1;
        while (j > 0) and (v < x[j]) do
        begin
            write('j=');
            write(j);
            write(' x[j]=');
            write(x[j]);
            writeln;
            x[j+1] := x[j];
            j := j-1
        end;
        write('j=');
        write(j);
        write(' x[j+1]:=');
        write(v);
        writeln;
        x[j+1] := v
    end;
    { Finished sorting, now print result }
    writeln;
    write('The integers, sorted:');
    writeln;
    writeln;
    h := 1;
    for h := 1 to 8 do
    begin
        write(x[h]);
        write('  ')
    end;
    writeln
end.
