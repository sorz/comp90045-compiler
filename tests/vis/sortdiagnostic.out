Provide 8 integers: h=2 x[h]=9
j=1 x[j+1]:=9
h=3 x[h]=1
j=2 x[j]=9
j=1 x[j]=5
j=0 x[j+1]:=1
h=4 x[h]=0
j=3 x[j]=9
j=2 x[j]=5
j=1 x[j]=1
j=0 x[j+1]:=0
h=5 x[h]=2
j=4 x[j]=9
j=3 x[j]=5
j=2 x[j+1]:=2
h=6 x[h]=8
j=5 x[j]=9
j=4 x[j+1]:=8
h=7 x[h]=100
j=6 x[j+1]:=100
h=8 x[h]=2
j=7 x[j]=100
j=6 x[j]=9
j=5 x[j]=8
j=4 x[j]=5
j=3 x[j+1]:=2

The integers, sorted:

0  1  2  2  5  8  9  100  
